ouroboros-network/sim-tests-lib/Test/Ouroboros/Network/Testnet.hs

```diff
@@ -3312,17 +3158,26 @@ unit_peer_sharing =
                                                           $ as))
                $ events'
 
-        events' =
-                 Trace.toList
+        events' = Trace.toList
                . splitWithNameTrace
                . fmap (\(WithTime t (WithName name b))
-                       -> (WithName name (WithTime t b)))
+                       -> WithName name (WithTime t b))
                . withTimeNameTraceEvents
                   @DiffusionTestTrace
                   @NtNAddr
-               . traceFromList
-               . fmap (\(t, tid, tl, te) -> SimEvent t tid tl te)
-               $ trace
+               -- We need roughly 1200 because:
+               -- * first peer sharing request will be issued after
+               --   `policyPeerSharAcitvationDelay = 300`
+               -- * this request will not bring any new peers, because non of the peers
+               --    are yet mature
+               -- * inbound connections become mature at 900s (15 mins)
+               -- * next peer share request happens after 900s, e.g. around 1200s.
+               . Trace.takeWhile (\se -> case se of
+                                          SimEvent    {seTime} -> seTime < Time 1250
+                                          SimPOREvent {seTime} -> seTime < Time 1250
+                                          _                    -> False
```

Comment:
We could redefine `SimEvent` type and cheat that there's not other constructor with pattern matches and `COMPLETE` pragma; the `SimRacesFound` constructor is internal to `IOSimPOR` if I recall correctly.

Reply:
I would hide it (since it makes a difference for all test authors), but this requires some care, e.g. if we define:
```
data SimEvent = SimeEvent_ {..} | SimPOREvent_ {..} | SimRacesFound_ {..}

pattern SimEvent = ...
pattern SimPOREvent = ...
{-# COMPLETE SimEvent, SimPOREvent #-}
```
Then we need to make sure that in `io-sim` use use the constructors not patterns (at least in `IOSimPOR`) to avoid strange bugs / warnings.

Reply:
Yes that would be better UI/UX but it should be done on io-sim side